src/main/java/org/reactome/addlinks/fileprocessors/XenbaseFileProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.reactome.addlinks.fileprocessors;
+
+import org.reactome.addlinks.EnsemblBioMartUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class XenbaseFileProcessor extends FileProcessor{
+
+    public XenbaseFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public XenbaseFileProcessor()
+    {
+        super();
+    }
+
+    private static final int uniprotIdentifierIndex = 0;
+    private static final int xenbaseIdentifierIndex = 3;
+
+    /**
+     * Build map of UniProt identifiers to Xenbase identifiers that is used to create Xenbase cross-references in database.
+     * @return - Map<String, List<String>
+     */
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFilePath = Paths.get(this.pathToFile.toAbsolutePath().toString().replace(".gz", ""));
+
```

Comment:
This looks similar to removal of the ".gz" and ".zip" prefix from path objects in the MGI and HPA classes.  A helper method to consolidate these where you pass in the path object and the suffix to be removed could be useful:

```
private static Path removeSuffix(Path filePath, String suffix) {
    return Paths.get(filePath.toAbsolutePath().toString().replace(suffix, ""));
}
```

Reply:
Done

Reply:
There was actually an even easier fix: Some of the files aren't gzipped when downloaded so I just removed that logic. Only VGNC (.gz) and HPA (.zip) have compressed download files